docs/src/components/productX/XHero.tsx

```diff
@@ -78,11 +78,10 @@ export default function XHero() {
     [],
   );
 
-  let rowGroupingCounter = 0;
+  const rowGroupingCounterRef = React.useRef(0);
   const isGroupExpandedByDefault = React.useCallback(() => {
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    rowGroupingCounter += 1;
-    return rowGroupingCounter === 3;
+    rowGroupingCounterRef.current += 1;
+    return rowGroupingCounterRef.current === 3;
```

Comment:
This is probably the only potentially sensitive change, but looks like the behavior is the same.

- Original: 
- PR: https://deploy-preview-42566--material-ui.netlify.app/x/

Reply:
The code, as it was before, seems very hacky. `isGroupExpandedByDefault` provides the node as an argument ([type here](https://github.com/mui/mui-x/blob/76832df0edb176ac4dbffb1650c35fc81e324d77/packages/x-data-grid/src/models/gridRows.ts#L128)), is there any info we could use to remove this hack?

If not possible, I think the functionality is the same and this looks like the proper adjustment 👍🏼

Reply:
I tried to get some useful info from the node but there's nothing stable that would let us know which is the 3rd group.

Reply:
Ok, no worries.